source/appModules/code.py

```diff
@@ -7,23 +7,41 @@
 
 import appModuleHandler
 import controlTypes
+from NVDAObjects.IAccessible.ia2Web import Editor
 from NVDAObjects.IAccessible.chromium import Document
+from NVDAObjects.IAccessible.ia2TextMozilla import MozillaCompoundTextInfo
 from NVDAObjects import NVDAObject, NVDAObjectTextInfo
 
 
+class VSCodeEditorTextInfo(MozillaCompoundTextInfo):
+	def _isCaretAtEndOfLine(self, caretObj) -> bool:
+		# #17039: IAccessibleText::textAtOffset with IA2_OFFSET_CARET is way too costly in VSCode.
+		# And doesn't actually work correctly in Chromium yet anyway.
```

Comment:
If this a problem in Chromium, shouldn't this apply to all Chromium text editors?

Reply:
True. But we'd really prefer that Chromium implement this correctly. And in the majority of cases this would not be a performance issue.  But just with VS Code it is very noticeable. If there is another scenario in Chromium where there is a performance hit we can move the code to be more broad.

Reply:
I suppose it would be a performance issue when using vscode.dev, but I have no idea how many people access that for anything but the smallest of changes.

Reply:
To be honest, Google internally uses a development environment based on vscode online. This will have a performance impact.

cc @derekriemer 
Could  you express your opinion on this?

I'm in favor of rolling this fix out to a wider scope until Chromium implements the interface.

If anyone can create a feature request in Chromium, I'd love to push it.